Valeriia/Movies_database_project/src/store/sagas/movieSaga.ts

```diff
@@ -0,0 +1,29 @@
+import {MovieActionTypes} from '../actions/actionTypes';
+import {all, call, put, takeLatest} from 'redux-saga/effects';
+import {fetchListOfMoviesBySortType, fetchMovieById} from './services';
+import {IRequestMovieAction, onRequestMovieSuccess, onRequestMovieError} from '../actions/movieActions';
+import { onRequestSuccessMovies } from '../actions/moviesAction';
+
+function* requestMovieSaga(action: IRequestMovieAction){
+    try{
+        const movie = yield call(fetchMovieById, action.id);
+        const moviesBySameGenre = yield call(fetchListOfMoviesBySortType, "release_date", "genres", movie.genres)
```

Comment:
:optional I would put(fetchMovies(....)) instead, so if there is an error, you will still show movie, but without moviesByGenre. In your case, if there is an error, you won't show movie